assignments/week_3/exercise_3_2/exercise_3_2.md

```diff
@@ -0,0 +1,151 @@
+
+    void add_to_front(std::string name);
+
+    void add_to_back(std::string name);
+
+    void add_at_index(std::string name, int index);
+
+    void remove_from_front();
+
+    void remove_from_back();
+
+    void remove_at_index(int index);
+
+    void print_names();
+
+};
+```
+
+Starter Code:
+```c++
+#include <string>
+#include <vector>
+#include <memory>
+
+class Node {
+public:
+    std::string name;
+    std::unique_ptr<Node> next;
+};
+
+class LinkedList {
+private:
+    std::unique_ptr<Node> head;
+public:
+
+    LinkedList();
+
+    void add_to_front(std::string name);
+
+    void add_to_back(std::string name);
+
+    void add_at_index(std::string name, int index);
+
+    void remove_from_front();
+
+    void remove_from_back();
+
+    void remove_at_index(int index);
+
+    void print_names();
+
+};
+```
+
+# Steps
+### add_to_front:
+1. Create a new variable that is a unique pointer to a Node using std::make_unique<>()
+2. Set the name of the new node to the name parameter passed into the function by changing the string variable called name to the passed in value by using -> to access the name member of the Node
+3. Set the new node’s next variable to the head pointer of the linked list by using std::move() on the head of the linked list
+4. Set the head of the linked list to the new node pointer again using std::move() except on the new node pointer
+### add_to_back:
+1. Create a new variable that is a unique pointer to a Node
+2. Set the name of new node pointer to be the name parameter and set its next pointer to nullptr
+3. Create an if else statement that checks if the linked list is empty by checking if the head is a nullptr
+4. Within the if statement create a variable, called curr, that is a pointer to a Node and set it to the head pointer using .get() on the head pointer
+5. Within the if statement create a while loop that sets curr to its next pointer while the next pointer of curr is not nullptr by using .get() on curr's next pointer
+6. Outside of the loop and inside the if statement set curr’s next pointer to be the new node pointer by using std::move()
+7. In the else statement set head to be the new node pointer again by using the std::move() function
+### add_at_index:
+1. Create a new variable that is a unique pointer to a Node
+2. Set the name of new node pointer to be the name parameter
+3. Create an if else statement that checks if the index is not 0
+4. In the if statement create a variable that is a pointer to a Node and define set it to the head pointer called curr using .get()
+5. In the if statement create a for loop that iterates index-1 number of times that sets curr to its next pointer using .get()
+6. Outside of the loop in the if statement set the new node’s next pointer to curr’s next pointer using std::move()
+7. In the if statement set curr’s next pointer to the new node pointer using std::move()
+8. In the else statement use the same technique in add to front to make the new node pointer the head
```

Comment:
I don't think you need to spell it out quite this much. Just link them to this page: https://visualgo.net/en/list